eng/pipelines/common/templates/runtimes/build-test-job.yml

```diff
@@ -21,6 +21,7 @@ parameters:
   pool: ''
   runtimeFlavor: 'coreclr'
   runtimeFlavorDisplayName: 'CoreCLR'
+  runtimeVariant: ''
```

Comment:
I think we should use "runtimeMode" like we are using for the interpreter:
https://github.com/dotnet/runtime/blob/863d4583f34b3d84010308edb3dd66721f7696ee/eng/pipelines/common/templates/runtimes/run-test-job.yml#L26

Reply:
Or just rename that use of runtimeMode to runtimeVariant, as long as they are the same.

Reply:
Is the interpreter being enabled something that'd show up as part of the `runtimeVariant`/`runtimeMode` string? "interpreter + llvm-statically-linked-into-mono-used-to-generate-IR-for-AOT-and-optionally-JIT + llc/opt-live-alongside-mono" is a perfectly valid configuration.

Reply:
Yes, Interpreter testing is being gated right now on `runtimeMode == 'interpreter'` - as you point out, interpreter plus LLVM is a valid combination, if `runtimeMode` and `runtimeVariant` are merged, that needs reconciling somehow. As does `runtimeVariant` having separate artifacts, but `runtimeMode` using vanilla artifacts.

Reply:
Well, my thought when I setup runtimeMode that way was that, while there are many combinations of things we could run, there are only a few we actually want to, and each one will have its own unique string. I'd use an enum if such a thing existed in yml.